code-challenges.test.js

```diff
@@ -32,15 +133,87 @@ const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
 const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
 // Expected output: [ 2, 1, -1 ]
 
+describe("remainderArray", () => {
+  it("returns a an array of only the remainders of the number when divided by 3", () => {
+    expect(remainderArray(hodgepodge1)).toEqual([ 2, 0, -1, 0 ])
+    expect(remainderArray(hodgepodge2)).toEqual([ 2, 1, -1 ])
+  })
+})
+
+//Test failed:  ReferenceError: remainderArray is not defined
+// Test Suites: 1 failed, 1 total
+// Tests:       1 failed, 1 passed, 2 total
+
+
 // b) Create the function that makes the test pass.
 
+//Pseudo:
+//Create a function called remainderArray.
+//Input: A mixed data array.
+//Iterate over array using .map
+//Check if each element is a number by using typeof
+//If element is a number, then use modulo by 3 to get the remainder
+//Use .filter to remove undefined values out of newArray 
+//Output: Returns an array of only the remainders of the number when divided by 3.
+
+const remainderArray = (array) => {
+  let newArray = []
+  newArray = array.map(value => {
+    if(typeof value === "number"){
+       return value % 3
+    }
+  })
+  return newArray.filter(value => 
+    value !== undefined)
+}
+
+console.log(remainderArray(hodgepodge1))
+console.log(remainderArray(hodgepodge2))
+
+//Test output: Passed
+// Test Suites: 1 passed, 1 total
+// Tests:       2 passed, 2 total
+
 // --------------------3) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
 
+
 // a) Create a test with an expect statement using the variables provided.
 
 const cubeAndSum1 = [2, 3, 4]
 // Expected output: 99
 const cubeAndSum2 = [0, 5, 10]
 // Expected output: 1125
 
+describe("cubed", () => {
+  it("returns the sum of all the numbers cubed", () => {
+    expect(cubed(cubeAndSum1)).toEqual(99)
+    expect(cubed(cubeAndSum2)).toEqual(1125)
+  })
+})
+
+//Test failed:  ReferenceError: cubed is not defined
+// Test Suites: 1 failed, 1 total
+// Tests:       1 failed, 2 passed, 3 total
+
 // b) Create the function that makes the test pass.
+
+//Pseudo:
+//Create a function called cubed.
+//Input: An array of numbers
+//Use .reduce method to accumulate the sum when each element number is cubed.
+//Output: The sum of all the numbers cubed.
+
+const cubed = (array) => {
+  //initial value is an optional argument, but needed for the reducer function in this case to know where to start in the array. 
+  //For some reason the output of cubedSum is 93 for the first text example without the initialValue set
+  let initialValue = 0
+  let cubedValues = array.reduce((cubedSum, currentValue) => cubedSum = cubedSum + (currentValue*currentValue*currentValue), initialValue)
+  return cubedValues
+}
+
+console.log(cubed(cubeAndSum1))
+console.log(cubed(cubeAndSum2))
```

Comment:
Since your function is invoked in your test, there is no need for console logs here.